daemon/create.go

```diff
@@ -77,6 +81,30 @@ func (daemon *Daemon) containerCreate(opts createOpts) (containertypes.Container
 		return containertypes.ContainerCreateCreatedBody{Warnings: warnings}, errdefs.InvalidParameter(err)
 	}
 
+	if img != nil && opts.params.Platform == nil {
+		p := platforms.DefaultSpec()
+
+		var platWarning bool
+		if img.OS != "" && img.OS != p.OS {
+			platWarning = true
+		}
+		if img.Architecture != "" && img.Architecture != p.Architecture {
+			platWarning = true
+		}
+		if img.Variant != "" && img.Variant != p.Variant {
+			platWarning = true
+		}
```

Comment:
I'm a little confused why this is matching all these bits directly -- shouldn't it instead just be using `platforms.Only` and asking if the image matches so it'll check the full vector for things like `linux/arm/v7` supporting `v6`, etc?

Essentially:

```go
if !platforms.Only(platforms.DefaultSpec()).Match(imagePlatform) {
// ... warning
}
```

Reply:
Fixed, also did this same thing for the image store.